File: app/Services/ProductService.php
<?php

namespace App\Services;

use App\Models\Product;

class ProductService
{
    public function getAllProducts()
    {
        return Product::all();
    }

    public function createProduct($data)
    {
        return Product::create($data);
    }

    public function getProductById($id)
    {
        return Product::findOrFail($id);
    }

    public function deleteProduct($id)
    {
        $product = $this->getProductById($id);
        $product->delete();
        return $product;
    }

    public function updateProduct($id, $data)
    {
        $product = $this->getProductById($id);
        $product->update($data);
        return $product;
    }
}

File: spring-project/src/main/java/com/example/demo/ProductService.java
package com.example.services;

import com.example.models.Product;
import com.example.repositories.ProductRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ProductService {

    private final ProductRepository productRepository;

    @Autowired
    public ProductService(ProductRepository productRepository) {
        this.productRepository = productRepository;
    }

    public List<Product> getAllProducts() {
        return productRepository.findAll();
    }

    public Product createProduct(Product data) {
        return productRepository.save(data);
    }

    public Product getProductById(Long id) {
        return productRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Product not found"));
    }

    public Product deleteProduct(Long id) {
        Product product = getProductById(id);
        productRepository.delete(product);
        return product;
    }

    public Product updateProduct(Long id, Product data) {
        Product product = getProductById(id);
        product.setName(data.getName());
        product.setPrice(data.getPrice());
        // Set other properties as needed
        return productRepository.save(product);
    }
}